internal/utils/config.go

```diff
@@ -130,7 +130,11 @@ func InitConfig(params InitParams, fsys afero.Fs) error {
 	c := config.NewConfig()
 	c.ProjectId = params.ProjectId
 	if params.UseOrioleDB {
-		c.Experimental.OrioleDBVersion = "15.1.0.150"
+		if c.Db.MajorVersion > 15 {
+			c.Experimental.OrioleDBVersion = "17.0.1.57"
+		} else {
+			c.Experimental.OrioleDBVersion = "15.1.0.150"
+		}
```

Comment:
There are some incompatibilities with orioledb 17. It's probably better to do this change separately https://github.com/supabase/cli/compare/develop...oriole-17